R/impact_central.R

```diff
@@ -332,16 +331,20 @@ impact_by_calendar_year <- function(baseline_burden, focal_burden) {
   assert_has_columns(focal_burden,
                      c("country", "burden_outcome", "year", "age", "value"))
 
-    baseline <- stats::aggregate(value ~ year + country + burden_outcome,
-                               baseline_burden,
-                               sum, na.rm = TRUE)
-  focal <- stats::aggregate(value ~ year + country + burden_outcome,
-                            focal_burden,
-                            sum, na.rm = TRUE)
-  data <- merge(baseline, focal,
-                c("country", "burden_outcome", "year"), sort = FALSE)
-  data$impact <- data$value.x - data$value.y
-  data[, c("country", "burden_outcome", "year", "impact")]
+  baseline <- baseline_burden %>%
+    dplyr::group_by(.data$year, .data$country, .data$burden_outcome) %>%
+    dplyr::summarise(baseline_val = sum(.data$value, na.rm = TRUE))
+  focal <- focal_burden %>%
+    dplyr::group_by(.data$year, .data$country, .data$burden_outcome) %>%
```

Comment:
The `.data` stuff is super ugly, I'm thinking maybe it is worth just having the horrible long R CMD check note instead

Reply:
yeah, there's no great way here. You can make the note go away by adding a shaddowing binding in the function itself:


```r
impact_by_calendar_year <- function(baseline_burden, focal_burden) {
  # ...
  year <- country <- burden_outcome <- value <- NULL
  baseline <- baseline_burden %>%
    dplyr::group_by(year, country, burden_outcome) %>%
    dplyr::summarise(baseline_val = sum(value, na.rm = TRUE))
  focal <- focal_burden %>%
    dplyr::group_by(year, country, burden_outcome) %>%
  # ...
```

which works like the `?globalVariables` function but will be local to the affected function (you'll still get the checking elsewhere)

Reply:
Yeah I like that, at least it isolates the ugliness